tests/test_admin/test_forms.py

```diff
@@ -420,6 +420,29 @@ def test_clean_discount(self, discount, is_valid, is_decimal_error):
                 error_message = 'Ensure that there are no more than 5 decimal places.'
             assert form.errors == {form.Fields.DISCOUNT: [error_message]}
 
+    @ddt.unpack
+    @ddt.data(
+        ("a thirst for knowledge", "a thirst for knowledge"),
+        ("   a thirst for knowledge   ", "a thirst for knowledge"),  # strips spaces
+        ("\r\t\n a thirst for knowledge", "a thirst for knowledge"),  # strips spaces
+        ("a thirst for knowledge\r\t\n ", "a thirst for knowledge"),  # strips spaces
+        ("    ", ""),
+    )
+    def test_clean_reason(self, reason, expected_reason):
+        form = self._make_bound_form("irrelevant@example.com", reason=reason)
+        assert form.is_valid()
+        cleaned_data = form.clean()
+        assert cleaned_data[ManageLearnersForm.Fields.REASON] == expected_reason
+
+    def test_validate_reason(self):
+        course_id = "course-v1:edX+DemoX+Demo_Course"
+        reason = ""
+        form = self._make_bound_form("irrelevant@example.com", course=course_id, reason=reason)
+        assert not form.is_valid()
+        assert form.errors == {
+            "__all__": [ValidationMessages.MISSING_REASON]
+        }
```

Comment:
@moconnell1453 
I think this test will not pass due to two reasons.
1. We need to mock `EnrollmentApiClient` as `clean_course` method is using this client to get details.
2. We also need to pass `course_mode` otherwise the form will raise validation error for course_mode too.
so this test will look like this.
```
    @mock.patch("enterprise.admin.forms.EnrollmentApiClient")
    def test_validate_reason(self, enrollment_client):
        instance = enrollment_client.return_value
        instance.get_course_details.side_effect = fake_enrollment_api.get_course_details
        course_id = "course-v1:edX+DemoX+Demo_Course"
        reason = ""
        form = self._make_bound_form("irrelevant@example.com", course=course_id, reason=reason, course_mode="audit")
        assert not form.is_valid()
        assert form.errors == {
            "__all__": [ValidationMessages.MISSING_REASON]
        }
```

Reply:
Ahmad, thank you for the feedback. I have been having some problems running unit tests locally. I will need to resolve those issues before I can effectively write and correct unit tests for edx-enterprise.